util/server/index.ts

```diff
@@ -0,0 +1,149 @@
+import type { Devfile, Host } from 'custom-types';
+import { promises as fs } from 'fs';
+import path from 'path';
+// @ts-expect-error js-yaml has no type definitions
+import { load as yamlToJSON } from 'js-yaml';
+
+interface getDevfileYAMLReturnType {
+  devfileYAML: string | null;
+  devfileJSON: Object | string | number | null | undefined;
+}
+
+const getConfigFileHosts = async (fileRelPath: string): Promise<Host> => {
+  const splitRelFilePath = fileRelPath.split('/');
+  const absFilePath = path.join(process.cwd(), ...splitRelFilePath);
+  const hostsUnparsed = await fs.readFile(absFilePath, 'utf8');
+  const hosts = JSON.parse(hostsUnparsed) as Host;
+  return hosts;
+};
+
+const getRemoteJSON = async (hostName: string, jsonLocation: string): Promise<Devfile[]> => {
+  const res = await fetch(`${jsonLocation}/index/all?icon=base64`);
+  const devfilesWithoutSource = (await res.json()) as Devfile[];
+  const devfilesWithSource = devfilesWithoutSource.map((devfile: Devfile) => {
+    devfile.sourceRepo = hostName;
+    return devfile;
+  });
+  return devfilesWithSource;
+};
+
+const getLocalJSON = async (hostName: string, jsonLocation: string): Promise<Devfile[]> => {
+  const devfileRelPath = jsonLocation.split('/');
+  const devfileAbsPath = path.join(process.cwd(), ...devfileRelPath, 'index.json');
```

Comment:
need to handle err cases, what if someone doesn't have an index.json?

Reply:
itll automatically throw an error saying file not found

Reply:
when `path.join` doesnt found a the location it throws an error.